x/gamm/pool-models/stableswap/pool.go

```diff
@@ -299,15 +315,16 @@ func (p *Pool) PokePool(blockTime time.Time) {}
 // It should only be able to be successfully called by the pool's ScalingFactorGovernor
 // TODO: move commented test for this function from x/gamm/keeper/pool_service_test.go once a pool_test.go file has been created for stableswap
 func (p *Pool) SetStableSwapScalingFactors(ctx sdk.Context, scalingFactors []uint64, scalingFactorGovernor string) error {
-	if scalingFactorGovernor != p.ScalingFactorGovernor {
+	if scalingFactorGovernor != p.ScalingFactorGovernor || len(scalingFactors) != p.PoolLiquidity.Len() {
```

Comment:
Non-blocking for this PR:

I think checking for governor here is not compatible with the abstraction.

I would like to suggest moving this governor check up to the caller. Then, we can also reuse `SetStableSwapScalingFactors` method in the constructor to reduce code duplication for checking scaling factor overflow

Reply:
The argument should be `sender` here, and this is the entry point of the `SetStableSwapScalingFactors` function right?